javascript/ql/src/semmle/javascript/Classes.qll

```diff
@@ -210,6 +217,23 @@ class ClassDeclStmt extends @classdeclstmt, ClassDefinition, Stmt {
  * A class expression.
  */
 class ClassExpr extends @classexpr, ClassDefinition, Expr {
+  override string getName() {
+    result = ClassDefinition.super.getName()
+    or
+    exists(VarDef vd | this = vd.getSource() | result = vd.getTarget().(VarRef).getName())
+    or
+    exists(Property p |
```

Comment:
Do we want to restrict this to value (i.e., non-accessor) properties?

Reply:
Sure. Shouldn't make a difference as the initializer of an accessor can't be a class expression.

Reply:
Ugh, sorry, that comment should have been against the corresponding code in `Functions.qll` (but changing both is fine, of course).

Reply:
It seems useful to have names for accessor functions.
I'm quite happy for the function in `{ get foo() {...} }` to have the name `foo`.

Reply:
OK. Both Spidermonkey and v8 seem to give it the name `get foo`, but I guess `foo` is fine.